### PR TITLE
refactor to move encapsulation of 'should we continue processing this…

### PR DESCRIPTION
**Pull request by mikerott**
… Nexpose Asset' out of the convert function, whose concern *should* just be to convert

### CODE REVIEW

File: pkg/assetfetcher/http.go [hunk 1]
```diff
@@ -88,6 +88,10 @@ func (c *NexposeAssetFetcher) FetchAssets(ctx context.Context, siteID string) (<
 	pagedErrChan := make(chan error, siteAssetResp.Page.TotalResources)
 
 	for _, asset := range siteAssetResp.Resources {
+		if !asset.hasBeenScanned() {
+			// no need to continue processing a Nexpose Asset that has never been scanned
+			continue
```
Comment (bkrebsbach) [inline pkg/assetfetcher/http.go]: Suggestion: Could you add a log and/or custom metric here so we have a sense of how many unscanned assets we're skipping? When we start to build out services to configure the scan sites in Nexpose, I could see this sort of information would very helpful.

Comment (mikerott) [reply pkg/assetfetcher/http.go]: I really like the custom metric idea.  I'll add it.

File: pkg/assetfetcher/config.go [hunk 2]
```diff
@@ -39,5 +41,6 @@ func (*AssetFetcherConfigComponent) New(_ context.Context, c *AssetFetcherConfig
 	return &NexposeAssetFetcher{
 		Host:     host,
 		PageSize: c.PageSize,
+		StatFn:   runhttp.StatFromContext,
```
Comment (bkrebsbach) [inline pkg/assetfetcher/config.go]: Nit: we've been avoiding using the runhttp library directly in code, trying to stick to using interfaces for loggers/staters so we can replace them more easily. The only place in code that references the specific runhttp package is in main.go. This also means we can generate mock loggers/staters for our tests. See some examples of this pattern from asset-inventory-api:
* https://github.com/asecurityteam/asset-inventory-api/blob/master/main.go#L27
* https://github.com/asecurityteam/asset-inventory-api/blob/master/pkg/domain/alias.go
* https://github.com/asecurityteam/asset-inventory-api/blob/master/pkg/handlers/v1/cloud_insert.go#L32
* https://github.com/asecurityteam/asset-inventory-api/blob/master/pkg/handlers/v1/stub_logger_test.go

Comment (bkrebsbach) [reply pkg/assetfetcher/config.go]: Here's a much better discussion about why to do this: https://github.com/asecurityteam/nexpose-asset-attributor/pull/1#discussion_r278641948

**Pull request merged**